core/coreunix/add.go

```diff
@@ -47,13 +47,31 @@ type Object struct {
 	Size  string
 }
 
+// A batch-adding dag service for use with importers (and only for that)
+// Always Commit() after adding.
+type batching struct {
```

Comment:
Unless we make the `DagBuilderHelper` take separately an `ipld.NodeAdder` and an `ipld.NodeGetter` (or an adder + a separate `DAGService`) rather than just a `DAGService`, we need to wrap Batch in a `DAGService`, as it seems `trickle` uses `Get`.

Reply:
If it uses get as I think it does, we probably need to commit before calling get. Let's go with plan b and make Batch implement DAGService (flushing on get). (unless you can think of a better solution)

Reply:
@Stebalien yeah i'll do that to be on the safe side.